rosbag2_transport/test/rosbag2_transport/test_rosbag2_node.cpp

```diff
@@ -125,6 +135,68 @@ TEST_F(RosBag2NodeFixture, publisher_and_subscriber_work)
   EXPECT_THAT(subscribed_messages[0], StrEq("Hello World"));
 }
 
+TEST_F(RosBag2NodeFixture, generic_subscription_uses_qos)
+{
+  // If the GenericSubscription does not use the provided QoS profile,
+  // its request will be incompatible with the Publisher's offer and no messages will be passed.
+  using namespace std::chrono_literals;
+  std::vector<std::string> test_messages = {"Hello World", "Hello World"};
```

Comment:
Shouldn't you test with different messages?

Reply:
I'll remove the contents check - all I care about is whether any message got through